### PR TITLE
fix(ui)#12833: soft delete service and edit action restrictions after soft delete

### LINKED ISSUE #12833: ui: add soft delete option for services and limit actions for soft deleted entities
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**

- [ ] Add soft delete option for services.
- [ ] Limit the actions for all the soft deleted entities. We should not allow actions such as edit owner, tier, etc. The entity details should have read-only view when soft deleted.

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes:

Fixes #12833 

I worked on adding edit action checks for soft-deleted entities for the following entities:
- [x] Services
- [x] Databases
- [x] Database Schemas
- [x] Tables
- [x] Topics
- [x] Dashboards
- [x] Pipelines
- [x] Containers
- [x] MlModels
- [x] Search Indices
- [x] Stored Procedures
- [x] Dashboard Data Modals

https://github.com/open-metadata/OpenMetadata/assets/51777795/92446e5d-7372-4e38-b9b2-f6eeff8fd44a

- [x] Teams

https://github.com/open-metadata/OpenMetadata/assets/51777795/a773a29b-2d60-4765-9f30-e4fdac32ab58

#
### Type of change:

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by harshach — APPROVED** [review]

**Comment by harshach** [comment]
@aniketkatkar97 can we add cypress tests for this

**Comment by aniketkatkar97** [comment]
@harshach Yes the tests are almost done. I will raise a separate PR for it.

File: openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx [hunk 1]
```diff
@@ -228,25 +249,27 @@ export const UserTab = ({
...
-                </UserSelectableList>
-              )}
-              <ManageButton
-                canDelete={false}
-                entityName={currentTeam.name}
-                extraDropdownContent={IMPORT_EXPORT_MENU_ITEM}
-              />
-            </Space>
-          </Col>
+          {!currentTeam.deleted && (
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx:252]:
```suggestion
          {!isTeamDeleted && (
```

File: openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx [hunk 2]
```diff
@@ -274,19 +297,20 @@ export const UserTab = ({
         )}
       </Col>
 
-      <ConfirmationModal
-        bodyText={t('message.are-you-sure-want-to-text', {
+      <Modal
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx:300]: Why did we replace confirmation modal here?

File: openmetadata-ui/src/main/resources/ui/src/components/common/DeleteWidget/DeleteWidgetModal.tsx [hunk 3]
```diff
@@ -219,45 +237,48 @@ const DeleteWidgetModal = ({
...
+            <Transi18next
+              i18nKey="message.type-delete-to-confirm"
+              renderElement={<strong />}
+            />
+          </div>
+
+          <Input
+            autoComplete="off"
+            data-testid="confirmation-text-input"
+            disabled={entityDeleteState.loading === 'waiting'}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DeleteWidget/DeleteWidgetModal.tsx:272]: Can we use autoFocus prop to make it focus instead manual code?

File: openmetadata-ui/src/main/resources/ui/src/pages/CustomPropertiesPageV1/CustomPropertiesPageV1.tsx [hunk 4]
```diff
@@ -126,7 +126,7 @@ const CustomEntityDetailV1 = () => {
         setIsButtonLoading(false);
       }
     },
-    [selectedEntityTypeDetail.id]
+    [selectedEntityTypeDetail]
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/CustomPropertiesPageV1/CustomPropertiesPageV1.tsx:129]: Why we change this? Can you add details of it?

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseDetailsPage/DatabaseDetailsPage.tsx [hunk 5]
```diff
@@ -171,37 +171,27 @@ const DatabaseDetails: FunctionComponent = () => {
     }
   };
 
-  const fetchDatabaseSchemas = (pagingObj?: string) => {
-    return new Promise<void>((resolve, reject) => {
-      setSchemaDataLoading(true);
-      getDatabaseSchemas(
+  const fetchDatabaseSchemas = async (pagingObj?: string) => {
+    try {
+      const response = await getDatabaseSchemas(
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseDetailsPage/DatabaseDetailsPage.tsx:176]:
```suggestion
     setSchemaDataLoading(true);
      const response = await getDatabaseSchemas(
```

File: openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/UserTab/UserTab.component.tsx:300]: I saw the ConfirmationModal component has been deprecated that's why I replaced it

File: openmetadata-ui/src/main/resources/ui/src/pages/CustomPropertiesPageV1/CustomPropertiesPageV1.tsx (hunk 4 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/CustomPropertiesPageV1/CustomPropertiesPageV1.tsx:129]: So we are using selectedEntityTypeDetail, as well inside the function, and the other data was not updating due to the specified dependancy

**Pull request merged**